tests/integration/workspace_access/test_secrets.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from datetime import timedelta
+
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.errors.mapping import NotFound
+from databricks.sdk.retries import retried
+from databricks.sdk.service.workspace import AclPermission
+
+from databricks.labs.ucx.workspace_access.groups import GroupManager
+from databricks.labs.ucx.workspace_access.manager import PermissionManager
+from databricks.labs.ucx.workspace_access.secrets import SecretScopesSupport
+
+logger = logging.getLogger(__name__)
+
+
+@retried(on=[NotFound], timeout=timedelta(minutes=3))
+def test_secrets(
+    ws: WorkspaceClient,
+    make_ucx_group,
+    sql_backend,
+    inventory_schema,
+    make_secret_scope,
+    make_secret_scope_acl,
+):
+    """
+    This test does the following:
+    * creates 2 ws groups and secret scopes
+    * assigns ACL permissions to the groups on the secrets
+    * run assessment
+    * migrate groups
+    * verify that the migrated groups has the same permissions on the scopes
+    :return:
+    """
+
+    created_items = []
+    for _ in range(2):
+        ws_group, acc_group = make_ucx_group()
+
+        scope = make_secret_scope()
+        make_secret_scope_acl(scope=scope, principal=ws_group.display_name, permission=AclPermission.WRITE)
+
+        scope_acl = ws.secrets.get_acl(scope, ws_group.display_name)
+        item = {
+            "ws_group": ws_group,
+            "acc_group": acc_group,
+            "scope": scope,
+            "scope_acl": scope_acl,
+        }
+        created_items.append(item)
+        logger.debug(f"secret scope acls: {scope_acl}")
+
+    created_groups_names = [i["ws_group"].display_name for i in created_items]
+
+    # Task 1 - crawl_permissions
```

Comment:
Can you please split this integration test the way that each task here is a separate/isolated integration test?

The reason is that we have them too flaky now and having bigger/longer tests decreases reliability.

Reply:
I'm not sure I understand. These tasks are dependent on each other: I cannot run Task 4's (apply_permissions_to_account_groups) `permission_manager.apply_group_permissions(migration_state)` without having run Task 1 (crawl_permissions)'s `pi.inventorize_permissions()`.

Reply:
Split this integration test into 4 different tests (for every task you do), so that the test runs faster because of the parallelism

Reply:
removed unnecessary tests and reduced the scope for the secrets permission integration test